RandBLAS/sparse.hh

```diff
@@ -170,29 +171,26 @@ static RNGState repeated_fisher_yates(
     for (int64_t j = 0; j < vec_len; ++j)
         vec_work[j] = j;
     std::vector<int64_t> pivots(vec_nnz);
-    typedef typename T_gen::ctr_type ctr_type;
-    ctr_type rout;
-    T_gen g;
-    RNGState out_state(init_state);
+    RNG gen;
+    auto [ctr, key] = state;
     for (int64_t i = 0; i < num_vecs; ++i) {
-        // set the state of the Random123 RNG.
         int64_t offset = i * vec_nnz;
-        Random123_RNGState<T_gen> impl_state(init_state);
-        impl_state.ctr.incr(offset);
+        auto ctri = ctr;
+        ctri.incr(offset);
         for (int64_t j = 0; j < vec_nnz; ++j) {
             // one step of Fisher-Yates shuffling
-            rout = g(impl_state.ctr, impl_state.key);
-            int64_t ell = j + rout.v[0] % (vec_len - j);
+            auto rv = gen(ctri, key);
+            int64_t ell = j + rv[0] % (vec_len - j);
```

Comment:
Did you write https://github.com/BallisticLA/RandBLAS/pull/28/files#diff-38df2d18a53b7af91f2aa92f6dc8b40a0582eae533acb874f8ba3ba92681cc6fR26-R33 so you could index into ``rv`` like this? Is ``gen`` still a Random123 RNG?